File: jlox/lox/lox/NativeClock.java
package lox;

import java.util.List;

class NativeClock implements LoxCallable {
    @Override
    public int arity() {
        return 0;
    }

    public Object call(Interpreter interpreter, List<Object> arguments) {
        return (double) System.currentTimeMillis() / 1000.0;
    }

    @Override
    public String toString() {
        return "<native fn clock>";
    }

}

File: jlox/plox/native_clock.py
import time
from plox.lox_callable import LoxCallable


class NativeClock(LoxCallable):
    def arity(self) -> int:
        return 0

    def call(self, interpreter, arguments: list):
        return time.time()

    def __str__(self) -> str:
        return "<native fn>"
